pkg/setting/setting.go

```diff
@@ -1988,16 +1988,17 @@ func (cfg *Cfg) readServerSettings(iniFile *ini.File) error {
 }
 
 // GetContentDeliveryURL returns full content delivery URL with /<edition>/<version> added to URL
-func (cfg *Cfg) GetContentDeliveryURL(prefix string) string {
-	if cfg.CDNRootURL != nil {
-		url := *cfg.CDNRootURL
-		preReleaseFolder := ""
-
-		url.Path = path.Join(url.Path, prefix, preReleaseFolder, cfg.BuildVersion)
-		return url.String() + "/"
+func (cfg *Cfg) GetContentDeliveryURL(prefix string) (string, error) {
+	if cfg.CDNRootURL == nil {
+		return "", nil
+	}
+	if cfg.BuildVersion == "" {
+		return "", errors.New("BuildVersion is not set")
```

Comment:
The semantics change, but I think it is OK -- and the docs are clear enough:
```

### cdn_url

{{% admonition type="note" %}}
Available in Grafana v7.4 and later versions.
{{% /admonition %}}

Specify a full HTTP URL address to the root of your Grafana CDN assets. Grafana will add edition and version paths.

For example, given a cdn url like `https://cdn.myserver.com` grafana will try to load a javascript file from
`http://cdn.myserver.com/grafana-oss/7.4.0/public/build/app.<hash>.js`.
```
https://github.com/grafana/grafana/blob/c11a76a34655c43d399ae6875bd92adc4b589a8b/docs/sources/setup-grafana/configure-grafana/_index.md?plain=1#L304